backend/store/etcd/watchers.go

```diff
@@ -79,51 +80,58 @@ func (s *Store) GetCheckConfigWatcher(ctx context.Context) <-chan store.WatchEve
 	return ch
 }
 
-// GetHookConfigWatcher returns a channel that emits WatchEventHookConfig structs notifying
-// the caller that a HookConfig was updated. If the watcher runs into a terminal error
+// GetEntityWatcher returns a channel that emits WatchEventEntity structs notifying
+// the caller that an Entity was updated. If the watcher runs into a terminal error
 // or the context passed is cancelled, then the channel will be closed. The caller must
 // restart the watcher, if needed.
-func (s *Store) GetHookConfigWatcher(ctx context.Context) <-chan store.WatchEventHookConfig {
-	ch := make(chan store.WatchEventHookConfig)
-	watcherChan := s.client.Watch(ctx, hookKeyBuilder.Build(""), clientv3.WithPrefix(), clientv3.WithCreatedNotify())
-
+func (s *Store) GetEntityWatcher(ctx context.Context) <-chan store.WatchEventEntity {
+	ch := make(chan store.WatchEventEntity, 1)
```

Comment:
I think this is a good choice, since it will allow the watcher to put a value there in the common case without waiting on a rendezvous with the consuming goroutine.